Hardware/CPU/AMD17CPU.cs

```diff
@@ -0,0 +1,515 @@
+          pwr = total_energy - _lastPwrValue;
+        else
+          pwr = (0xffffffff - _lastPwrValue) + total_energy;
+
+        // update for next sample 
+        _lastPwrTime = sample_time;
+        _lastPwrValue = total_energy;
+
+        double energy = 15.3e-6 * pwr;
+        energy /= time.TotalSeconds;
+
+        PackagePower.Value = (float)energy;
+        _hw.ActivateSensor(PackagePower);
+
+        // current temp Bit [31:21] 
+        temperature = (temperature >> 21) * 125;
+        float offset = 0.0f;
+        if (cpu.Name != null && (cpu.Name.Contains("1600X") || cpu.Name.Contains("1700X") || cpu.Name.Contains("1800X")))
+          offset = -20.0f;
+        else if (cpu.Name != null && (cpu.Name.Contains("1920X") || cpu.Name.Contains("1950X")))
+          offset = -27.0f;
+        else if (cpu.Name != null && (cpu.Name.Contains("1910") || cpu.Name.Contains("1920")))
+          offset = -10.0f;
+
+        CoreTemperatureTctl.Value = (temperature * 0.001f);
+        CoreTemperatureTdie.Value = (temperature * 0.001f) + offset;
+        _hw.ActivateSensor(CoreTemperatureTctl);
+        _hw.ActivateSensor(CoreTemperatureTdie);
+
+        // voltage 
+        double VIDStep = 0.00625;
+        double vcc;
+        uint svi0_plane_x_vddcor;
+        uint svi0_plane_x_iddcor;
+
+        //Core
+        if ((smusvi0_tfn & 0x01) == 0)
+        {
+          svi0_plane_x_vddcor = (smusvi0_tel_plane0 >> 16) & 0xff;
+          svi0_plane_x_iddcor = smusvi0_tel_plane0 & 0xff;
+          vcc = 1.550 - (double)VIDStep * svi0_plane_x_vddcor;
+          CoreVoltage.Value = (float)vcc;
+          _hw.ActivateSensor(CoreVoltage);
+          // coreCurrent.Value = (float)(svi0_plane_x_iddcor * 1); 
+          // hw.ActivateSensor(coreCurrent); 
+        }
+
+        // SoC 
+        if ((smusvi0_tfn & 0x02) == 0)
+        {
+          svi0_plane_x_vddcor = (smusvi0_tel_plane1 >> 16) & 0xff;
+          svi0_plane_x_iddcor = smusvi0_tel_plane1 & 0xff;
+          vcc = 1.550 - (double)VIDStep * svi0_plane_x_vddcor;
+          SocVoltage.Value = (float)vcc;
+          _hw.ActivateSensor(SocVoltage);
+          //socCurrent.Value = (float)(svi0_plane_x_iddcor * 1);
+          //hw.ActivateSensor(socCurrent);
+        }
+
+      }
+      #endregion
+
+      public void AppendThread(CPUID thread, int numa_id, int core_id)
+      {
+        NumaNode node = null;
+        // node = (from x in nodes 
+        //        where x.nodeId == numa_id 
+        //        select x).FirstOrDefault(); 
+        foreach (var n in Nodes)
+        {
+          if (n.NodeId == numa_id)
+            node = n;
+        }
+        if (node == null)
+        {
+          node = new NumaNode(_hw);
+          node.NodeId = numa_id;
+          node.Parent = this;
+          Nodes.Add(node);
+        }
+
+        if (thread != null)
+          node.AppendThread(thread, core_id);
+      }
+
+      public Sensor CoreTemperatureTctl { get; set; }
```

Comment:
All sensors should be private (same with the sensors below). I don't think there's any hardware in OHM that has explicit sensors added like this.